File: API/src/main/java/com/concierge/apiconcierge/models/workshop/toolcontrol/ToolControlRequest.java
package com.concierge.apiconcierge.models.workshop.toolcontrol;

import com.concierge.apiconcierge.models.workshop.toolcontrol.enums.StatusRequest;
import com.concierge.apiconcierge.models.workshop.toolcontrol.enums.TypeCategory;
import com.concierge.apiconcierge.models.workshop.toolcontrol.enums.TypeRequest;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
@Entity
@Table(name = "tb_tool_control_request")
public class ToolControlRequest {

    @Column(name = "company_id")
    private Integer companyId;

    @Column(name = "resale_id")
    private Integer resaleId;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private StatusRequest status;

    @Column(name = "request_type")
    private TypeRequest requestType;

    @Column(name = "request_date")
    private Date requestDate;

    @Column(name = "request_information")
    private String requestInformation;

    @Column(name = "request_user_id")
    private Integer requestUserId;

    @Column(name = "request_user_name")
    private String requestUserName;

    @Column(name = "category_type")
    private TypeCategory categoryType;

    @Column(name = "mechanic_id")
    private Integer mechanicId;
}

File: Front-End/src/app/models/workshop/tool.control/tool.control.request.ts
import { StatusTollControlRequestEnum } from "../status.tool.control.request.enum";
import { TypeCategoryEnum } from "../type.category.enum";
import { TypeMaterialEnum } from "../type.material.enum";

export class ToolControlRequest {
    companyId: number | null = null;
    resaleId: number | null = null;
    id: number | null = null;
    status: StatusTollControlRequestEnum = StatusTollControlRequestEnum.OPEN;
    requestType: TypeMaterialEnum = TypeMaterialEnum.BOTH;
    requestDate: string = "";
    requestInformation: string = "";
    requestUserId: number | null = null;
    requestUserName: string = "";
    categoryType: TypeCategoryEnum = TypeCategoryEnum.OTHER;
    mechanicId: number | null = null;
}